libs/community/langchain_community/chat_models/huggingface.py

```diff
@@ -71,13 +80,50 @@ def validate_llm(cls, values: dict) -> dict:
             )
         return values
 
+    def _stream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[ChatGenerationChunk]:
+        request = self._to_chat_prompt(messages)
+
+        for data in self.llm._stream(request, **kwargs):
+            delta = data.text
+            chunk = ChatGenerationChunk(message=AIMessageChunk(content=delta))
+            if run_manager:
+                run_manager.on_llm_new_token(delta, chunk=chunk)
+            yield chunk
+
+    async def _astream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[ChatGenerationChunk]:
+        request = self._to_chat_prompt(messages)
+        async for data in self.llm._astream(request, **kwargs):  # await 대신 async for 사용
+            delta = data.text
+            chunk = ChatGenerationChunk(message=AIMessageChunk(content=delta))
+            if run_manager:
+                await run_manager.on_llm_new_token(delta, chunk=chunk)
+            yield chunk
+
     def _generate(
         self,
         messages: List[BaseMessage],
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
         **kwargs: Any,
     ) -> ChatResult:
+        if self.streaming:
```

Comment:
we generally don't want to support self`.streaming ` on _generate going forward -- we want users to use .stream() or .astream() for streaming exclusively

Reply:
@eyurtsev 
I've updated the code based on lint recommendations. 😊
However, the discussion about removing self.streaming=True might need further consideration. 😢

Notably, ChatOpenAI supports streaming=True here:
(https://github.com/langchain-ai/langchain/blob/53ac1ebbbccbc16ce57badf08522c6e59256fdfe/libs/community/langchain_community/chat_models/openai.py#L150)

As for LLMChain, it currently requires the addition of streaming support:
(https://github.com/langchain-ai/langchain/blob/53ac1ebbbccbc16ce57badf08522c6e59256fdfe/libs/langchain/langchain/chains/llm.py#L37)

I'll be looking forward to any updates on this topic and am ready to follow up accordingly.